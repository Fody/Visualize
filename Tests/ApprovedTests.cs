using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

[UseReporter(typeof(DiffReporter))]
#if(DEBUG)
[UseApprovalSubdirectory("results-debug")]
#else
[UseApprovalSubdirectory("results-release")]
#endif
[TestFixture]
public class ApprovedTests
{
    [Test]
    public void ClassWithProperties()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithProperties"));
    }

    [Test]
    public void ClassWithConst()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithConst"));
    }

    [Test]
    public void ClassWithIndexor()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIndexor"));
    }

    [Test]
    public void ClassWithDataAnnotations()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithDataAnnotations"));
    }

    [Test]
    public void ClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerable"));
    }

    [Test]
    public void ClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithICollection"));
    }

    [Test]
    public void GenericClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithIEnumerable`1"));
    }

    [Test]
    public void GenericClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithICollection`1"));
    }

    [Test]
    public void ClassWithIEnumerableNotAttributed()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerableNotAttributed"));
    }

    //[Test]
    //public void Sample()
    //{
    //    var sampleClassType = AssemblyWeaver.Assembly.GetType("AssemblyToProcess.ClassWithIEnumerable");
    //    var sample = Activator.CreateInstance(sampleClassType);
    //}
}