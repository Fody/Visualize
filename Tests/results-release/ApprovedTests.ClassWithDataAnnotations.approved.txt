.class public auto ansi beforefieldinit AssemblyToProcess.ClassWithDataAnnotations
       extends [mscorlib]System.Object
{
  .custom instance void [mscorlib]System.Diagnostics.DebuggerDisplayAttribute::.ctor(string) = ( 01 00 16 7B 44 65 62 75 67 67 65 72 44 69 73 70   // ...{DebuggerDisp
                                                                                                 6C 61 79 28 29 2C 6E 71 7D 00 00 )                // lay(),nq}..
  .field private string '<LastName>k__BackingField'
  .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
  .field private string '<FirstName>k__BackingField'
  .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
  .method public hidebysig specialname instance string 
          get_LastName() cil managed
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    // Code size       7 (0x7)
    .maxstack  8
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    .line 8,8 : 34,38 'AssemblyToProcess\\ClassWithDataAnnotations.cs'
    IL_0000:  ldarg.0
    IL_0001:  ldfld      string AssemblyToProcess.ClassWithDataAnnotations::'<LastName>k__BackingField'
    IL_0006:  ret
  } // end of method ClassWithDataAnnotations::get_LastName
  .method public hidebysig specialname instance void 
          set_LastName(string 'value') cil managed
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    // Code size       8 (0x8)
    .maxstack  8
    .line 8,8 : 39,43 ''
    IL_0000:  ldarg.0
    IL_0001:  ldarg.1
    IL_0002:  stfld      string AssemblyToProcess.ClassWithDataAnnotations::'<LastName>k__BackingField'
    IL_0007:  ret
  } // end of method ClassWithDataAnnotations::set_LastName
  .method public hidebysig specialname instance string 
          get_FirstName() cil managed
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    // Code size       7 (0x7)
    .maxstack  8
    .line 11,11 : 35,39 ''
    IL_0000:  ldarg.0
    IL_0001:  ldfld      string AssemblyToProcess.ClassWithDataAnnotations::'<FirstName>k__BackingField'
    IL_0006:  ret
  } // end of method ClassWithDataAnnotations::get_FirstName
  .method public hidebysig specialname instance void 
          set_FirstName(string 'value') cil managed
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    // Code size       8 (0x8)
    .maxstack  8
    .line 11,11 : 40,44 ''
    IL_0000:  ldarg.0
    IL_0001:  ldarg.1
    IL_0002:  stfld      string AssemblyToProcess.ClassWithDataAnnotations::'<FirstName>k__BackingField'
    IL_0007:  ret
  } // end of method ClassWithDataAnnotations::set_FirstName
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } // end of method ClassWithDataAnnotations::.ctor
  .method private hidebysig specialname instance string 
          DebuggerDisplay() cil managed
  {
    // Code size       37 (0x25)
    .maxstack  4
    .locals init (object[] V_0)
    IL_0000:  ldstr      "Last Name = \"{0}\" | First Name = \"{1}\""
    IL_0005:  ldc.i4.2
    IL_0006:  newarr     [mscorlib]System.Object
    IL_000b:  stloc.0
    IL_000c:  ldloc.0
    IL_000d:  ldc.i4.0
    IL_000e:  ldarg.0
    IL_000f:  call       instance string AssemblyToProcess.ClassWithDataAnnotations::get_LastName()
    IL_0014:  stelem.ref
    IL_0015:  ldloc.0
    IL_0016:  ldc.i4.1
    IL_0017:  ldarg.0
    IL_0018:  call       instance string AssemblyToProcess.ClassWithDataAnnotations::get_FirstName()
    IL_001d:  stelem.ref
    IL_001e:  ldloc.0
    IL_001f:  call       string [mscorlib]System.String::Format(string,
                                                                object[])
    IL_0024:  ret
  } // end of method ClassWithDataAnnotations::DebuggerDisplay
  .property instance string LastName()
  {
    .custom instance void [System.ComponentModel.DataAnnotations]System.ComponentModel.DataAnnotations.DisplayAttribute::.ctor() = ( 01 00 02 00 54 0E 04 4E 61 6D 65 09 4C 61 73 74   // ....T..Name.Last
                                                                                                                                     20 4E 61 6D 65 54 08 05 4F 72 64 65 72 00 00 00   //  NameT..Order...
                                                                                                                                     00 ) 
    .get instance string AssemblyToProcess.ClassWithDataAnnotations::get_LastName()
    .set instance void AssemblyToProcess.ClassWithDataAnnotations::set_LastName(string)
  } // end of property ClassWithDataAnnotations::LastName
  .property instance string FirstName()
  {
    .custom instance void [System.ComponentModel.DataAnnotations]System.ComponentModel.DataAnnotations.DisplayAttribute::.ctor() = ( 01 00 02 00 54 0E 04 4E 61 6D 65 0A 46 69 72 73   // ....T..Name.Firs
                                                                                                                                     74 20 4E 61 6D 65 54 08 05 4F 72 64 65 72 01 00   // t NameT..Order..
                                                                                                                                     00 00 ) 
    .get instance string AssemblyToProcess.ClassWithDataAnnotations::get_FirstName()
    .set instance void AssemblyToProcess.ClassWithDataAnnotations::set_FirstName(string)
  } // end of property ClassWithDataAnnotations::FirstName
} // end of class AssemblyToProcess.ClassWithDataAnnotations